android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -57,24 +58,27 @@ public static AppcoinsBillingStubHelper getInstance() {
   }
 
   @Override public int isBillingSupported(int apiVersion, String packageName, String type) {
-
-    if (WalletUtils.hasWalletInstalled()) {
-      try {
-        return serviceAppcoinsBilling.isBillingSupported(apiVersion, packageName, type);
-      } catch (RemoteException e) {
-        e.printStackTrace();
-        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
-      }
-    } else {
-      if (type.equalsIgnoreCase("inapp")) {
-        if (apiVersion == SUPPORTED_API_VERSION) {
-          return ResponseCode.OK.getValue();
+    if (isDeviceVersionSupported()) {
```

Comment:
did you consider using something like:
```suggestion    
    if (!isDeviceVersionSupported()) {
      return ResponseCode.BILLING_UNAVAILABLE.getValue();
    }  
```
instead of creating other big if?